File: Daily Test/C/Water Buckets 27.4.2025.c
#include <stdio.h>

int main() {
    int n, x;
    scanf("%d", &n);
    
    int buckets[n];
    for (int i = 0; i < n; i++) {
        scanf("%d", &buckets[i]);
    }
    scanf("%d", &x);
    
    for (int i = n - 1; i >= 0; i--) {
        if (buckets[i] > x) {
            int leak = buckets[i] - x;
            buckets[i] = x;
            if (i > 0) {
                buckets[i - 1] += leak;
            }
        }
    }
    
    for (int i = 0; i < n; i++) {
        printf("%d ", buckets[i]);
    }
    
    return 0;
}

File: Daily Test/Python/Water Buckets 27.4.2025.py
n = int(input())
buckets = list(map(int, input().split()))
x = int(input())

for i in range(n-1, -1, -1):
    if buckets[i] > x:
        leak = buckets[i] - x
        buckets[i] = x
        if i > 0:
            buckets[i-1] += leak

print(*buckets)
